node/src/tasks/bsp_upload_file_task.rs

```diff
@@ -0,0 +1,339 @@
+use std::{path::Path, str::FromStr};
+
+use anyhow::anyhow;
+use forest_manager::traits::ForestStorage;
+use log::debug;
+use sc_network::PeerId;
+use sc_tracing::tracing::{error, info, warn};
+use shc_common::types::{FileKey, FileMetadata, HasherOutT};
+
+use file_manager::traits::{FileStorage, FileStorageWriteError, FileStorageWriteOutcome};
+use sp_core::H256;
+use sp_trie::TrieLayout;
+use storage_hub_infra::{actor::ActorHandle, event_bus::EventHandler};
+use tokio::{fs::File, io::AsyncWriteExt};
+
+use crate::services::{
+    blockchain::{
+        commands::BlockchainServiceInterface, events::NewStorageRequest,
+        handler::BlockchainService, types::ExtrinsicResult,
+    },
+    file_transfer::{commands::FileTransferServiceInterface, events::RemoteUploadRequest},
+    handler::{StorageHubHandler, StorageHubHandlerConfig},
+};
+
+const LOG_TARGET: &str = "bsp-upload-file-task";
+
+pub struct BspUploadFileTask<SHC: StorageHubHandlerConfig> {
+    storage_hub_handler: StorageHubHandler<SHC>,
+    file_key_cleanup: Option<HasherOutT<SHC::TrieLayout>>,
+}
+
+impl<SHC: StorageHubHandlerConfig> Clone for BspUploadFileTask<SHC> {
+    fn clone(&self) -> BspUploadFileTask<SHC> {
+        Self {
+            storage_hub_handler: self.storage_hub_handler.clone(),
+            file_key_cleanup: self.file_key_cleanup.clone(),
+        }
+    }
+}
+
+impl<SHC: StorageHubHandlerConfig> BspUploadFileTask<SHC> {
+    pub fn new(storage_hub_handler: StorageHubHandler<SHC>) -> Self {
+        Self {
+            storage_hub_handler,
+            file_key_cleanup: None,
+        }
+    }
+}
+
+impl<SHC: StorageHubHandlerConfig> EventHandler<RemoteUploadRequest> for BspUploadFileTask<SHC>
```

Comment:
Could be worth to comment what part of the flow of this task this handler is doing.